autofit/optimize/non_linear/downhill_simplex.py

```diff
@@ -18,7 +18,7 @@ def __init__(
             fmin=scipy.optimize.fmin
     ):
         if paths is None:
-            paths = Paths()
+            paths = Paths(non_linear_name=self.name)
```

Comment:
Interesting. Should we maybe be using polymorphism here?

Reply:
This now uses the class names - there may be a better refactor going forward.